From bbe3fbd5cb16007b99cbf492118d32c6831b9fda Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Tue, 14 Jun 2022 13:12:10 +0100
Subject: [PATCH 1/3] Remove the Pine64 Quartz copy of the RPi NVRAM

We can't just copy the Rasperry Pi NVRAM, their modules are custom
made so they have their own special NVRAMs for their variant. While
they may work for other modules it could put them out of compliance
or worse. We need to add an appropraite AzureWave AW-CM256SM NVRAM.

Even if it was the right one it should have just been linked rather
than actually copying it again too.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 WHENCE                                        |  1 -
 brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt | 83 -------------------
 2 files changed, 84 deletions(-)
 delete mode 100644 brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt

diff --git a/WHENCE b/WHENCE
index 776dd8b..cc37751 100644
--- a/WHENCE
+++ b/WHENCE
@@ -2952,7 +2952,6 @@ File: "brcm/brcmfmac43430-sdio.MUR1DX.txt"
 File: "brcm/brcmfmac43430-sdio.raspberrypi,3-model-b.txt"
 Link: brcm/brcmfmac43430-sdio.raspberrypi,model-zero-w.txt -> brcmfmac43430-sdio.raspberrypi,3-model-b.txt
 File: "brcm/brcmfmac43455-sdio.acepc-t8.txt"
-File: "brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt"
 File: "brcm/brcmfmac43455-sdio.raspberrypi,3-model-b-plus.txt"
 Link: brcm/brcmfmac43455-sdio.raspberrypi,3-model-a-plus.txt -> brcmfmac43455-sdio.raspberrypi,3-model-b-plus.txt
 File: "brcm/brcmfmac43455-sdio.raspberrypi,4-model-b.txt"
diff --git a/brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt b/brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt
deleted file mode 100644
index abaf544..0000000
--- a/brcm/brcmfmac43455-sdio.pine64,quartz64-b.txt
+++ /dev/null
@@ -1,83 +0,0 @@
-# SPDX-License-Identifier: GPL-2.0+
-# (C) Copyright 2018 Raspberry Pi (Trading) Ltd.
-# NVRAM config file for the BCM43455 WiFi/BT chip as found on the
-# Pine64 Quartz64 Model B
-aa2g=1
-aa5g=1
-AvVmid_c0=1,165,2,100,2,100,2,100,2,100
-boardflags=0x00480201
-boardflags2=0x40800000
-boardflags3=0x44200100
-boardrev=0x1304
-boardtype=0x6e4
-btc_mode=1
-btc_params1=0x7530
-btc_params8=0x4e20
-btc_params50=0x972c
-cbfilttype=1
-cckPwrIdxCorr=3
-cckTssiDelay=150
-deadman_to=481500000
-devid=0x43ab
-dot11agofdmhrbw202gpo=0x4442
-dot11b_opts=0x3aa85
-ed_thresh2g=-54
-ed_thresh5g=-54
-eu_edthresh2g=-54
-eu_edthresh5g=-54
-extpagain2g=2
-extpagain5g=2
-fdsslevel_ch11=6
-femctrl=0
-itrsw=1
-ldo1=4
-ltecxfnsel=0x22
-ltecxgcigpio=0x32
-ltecxmux=0
-ltecxpadnum=0x0504
-macaddr=b8:27:eb:74:f2:6c
-manfid=0x2d0
-maxp2ga0=80
-maxp5ga0=82,82,82,82
-mcsbw202gpo=0x98444422
-mcsbw205ghpo=0xb9555000
-mcsbw205glpo=0xb9555000
-mcsbw205gmpo=0xb9555000
-mcsbw402gpo=0x98444422
-mcsbw405ghpo=0xb9555000
-mcsbw405glpo=0xb9555000
-mcsbw405gmpo=0xb9555000
-mcsbw805ghpo=0xb9555000
-mcsbw805glpo=0xb9555000
-mcsbw805gmpo=0xb9555000
-nocrc=1
-ofdmlrbw202gpo=0x0022
-ofdmTssiDelay=150
-pa2ga0=-129,6525,-718
-pa2ga1=-149,4408,-601
-pa5ga0=-185,6836,-815,-186,6838,-815,-184,6859,-815,-184,6882,-818
-pa5ga1=-202,4285,-574,-201,4312,-578,-196,4391,-586,-201,4294,-575
-pdoffset2g40ma0=16
-pdoffset2gperchan=0,-2,1,0,1,0,1,1,1,0,0,-1,-1,0
-pdoffset40ma0=0x8888
-pdoffset80ma0=0x8888
-pdoffsetcckma0=2
-phycal_tempdelta=15
-prodid=0x06e4
-rawtempsense=0x1ff
-rxchain=1
-sromrev=11
-swctrlmap_2g=0x00000000,0x00000000,0x00000000,0x010000,0x3ff
-swctrlmap_5g=0x00100010,0x00200020,0x00200020,0x010000,0x3fe
-swctrlmapext_2g=0x00000000,0x00000000,0x00000000,0x000000,0x3
-swctrlmapext_5g=0x00000000,0x00000000,0x00000000,0x000000,0x3
-tssipos2g=1
-tssipos5g=1
-tworangetssi2g=1
-tworangetssi5g=1
-txchain=1
-txpwr2gAdcScale=1
-txpwr5gAdcScale=1
-vcodivmode=1
-vendid=0x14e4
-xtalfreq=37400
-- 
2.36.1

From 06a63595b23414abd00321be732837cb231a313c Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Tue, 14 Jun 2022 13:40:06 +0100
Subject: [PATCH 2/3] Add initial AzureWave AW-CM256SM NVRAM file

The AzureWave AW-CM256SM wireless module is a WiFi/BT combo
module built using the Cypress CYW43455. Add an initial
NVRAM based on the one provided for the Beagle-AI which
contains this WiFi module. We can update/improve this easily
enough if we get a newer rev of the NVRAM later.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 WHENCE                                 |  1 +
 brcm/brcmfmac43455-sdio.AW-CM256SM.txt | 66 ++++++++++++++++++++++++++
 2 files changed, 67 insertions(+)
 create mode 100644 brcm/brcmfmac43455-sdio.AW-CM256SM.txt

diff --git a/WHENCE b/WHENCE
index cc37751..084af41 100644
--- a/WHENCE
+++ b/WHENCE
@@ -2963,6 +2963,7 @@ File: brcm/brcmfmac4356-sdio.AP6356S.txt
 Link: brcm/brcmfmac4356-sdio.firefly,firefly-rk3399.txt -> brcmfmac4356-sdio.AP6356S.txt
 Link: brcm/brcmfmac4356-sdio.khadas,vim2.txt -> brcmfmac4356-sdio.AP6356S.txt
 Link: brcm/brcmfmac4356-sdio.vamrs,rock960.txt -> brcmfmac4356-sdio.AP6356S.txt
+File: brcm/brcmfmac43455-sdio.AW-CM256SM.txt
 
 Licence: GPLv2. See GPL-2 for details.
 
diff --git a/brcm/brcmfmac43455-sdio.AW-CM256SM.txt b/brcm/brcmfmac43455-sdio.AW-CM256SM.txt
new file mode 100644
index 0000000..4ca98ee
--- /dev/null
+++ b/brcm/brcmfmac43455-sdio.AW-CM256SM.txt
@@ -0,0 +1,66 @@
+aa2g=1
+aa5g=1
+AvVmid_c0=0,157,1,126,1,126,1,126,1,126
+boardflags=0x00080201
+boardflags2=0x40000000
+boardflags3=0x48200100
+boardrev=0x1304
+boardtype=0x6e4
+deadman_to=481500000
+devid=0x43ab
+dot11agofdmhrbw202gpo=0x1553
+ed_thresh2g=-54
+ed_thresh5g=-54
+extpagain2g=2
+extpagain5g=2
+femctrl=0
+itrsw=1
+ltecxfnsel=0x22
+ltecxgcigpio=0x32
+ltecxmux=0
+ltecxpadnum=0x0504
+macaddr=00:90:4c:c5:12:38
+manfid=0x2d0
+maxp2ga0=64
+maxp5ga0=80,82,76,77
+mcsbw202gpo=0x99355533
+mcsbw205ghpo=0x99855000
+mcsbw205glpo=0x99755000
+mcsbw205gmpo=0x9df55000
+mcsbw405ghpo=0xd9755000
+mcsbw405glpo=0xb8555000
+mcsbw405gmpo=0xed955000
+mcsbw805ghpo=0xd9555000
+mcsbw805glpo=0xc8555000
+mcsbw805gmpo=0xe9555000
+muxenab=0x10
+nocrc=1
+ofdmlrbw202gpo=0x0033
+pa2ga0=-112,6296,-662
+pa2ga1=-165,3699,-515
+pa5ga0=-143,6016,-683,-141,6013,-678,-137,5988,-670,-136,5982,-670
+pa5ga1=-161,3544,-499,-166,3543,-497,-169,3569,-497,-171,3598,-498
+pdoffset2g40ma0=10
+pdoffset40ma0=0xaaaa
+pdoffset80ma0=0xaaaa
+prodid=0x06e4
+rxchain=1
+rxgains2gelnagaina0=0
+rxgains2gtrelnabypa0=0
+rxgains2gtrisoa0=0
+rxgains5gelnagaina0=0
+rxgains5gtrelnabypa0=0
+rxgains5gtrisoa0=0
+sromrev=11
+swctrlmap_2g=0x00040004,0x00020002,0x00040004,0x010a02,0x1ff
+swctrlmap_5g=0x00100010,0x00200020,0x00100010,0x010a02,0x2f4
+swctrlmapext_2g=0x00000000,0x00000000,0x00000000,0x000000,0x000
+swctrlmapext_5g=0x00000000,0x00000000,0x00000000,0x000000,0x000
+tssipos2g=1
+tssipos5g=1
+tworangetssi2g=1
+tworangetssi5g=1
+txchain=1
+vcodivmode=1
+vendid=0x14e4
+xtalfreq=37400
-- 
2.36.1

From e472819be4a5aa4826a557e2ec92ce3116f9566d Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Tue, 14 Jun 2022 15:09:00 +0100
Subject: [PATCH 3/3] Link some devices that ship with the AW-CM256SM

We know the BeagleBoneAI, Pine64 Pinebook Pro, Pinephone Pro and
Pine64 Quartz64 Model B all ship with a AW-CM256SM so link their
known names so the WiFi NVRAM is available to those devices.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 WHENCE | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/WHENCE b/WHENCE
index 084af41..ae13c5d 100644
--- a/WHENCE
+++ b/WHENCE
@@ -2964,6 +2964,10 @@ Link: brcm/brcmfmac4356-sdio.firefly,firefly-rk3399.txt -> brcmfmac4356-sdio.AP6
 Link: brcm/brcmfmac4356-sdio.khadas,vim2.txt -> brcmfmac4356-sdio.AP6356S.txt
 Link: brcm/brcmfmac4356-sdio.vamrs,rock960.txt -> brcmfmac4356-sdio.AP6356S.txt
 File: brcm/brcmfmac43455-sdio.AW-CM256SM.txt
+Link: brcm/brcmfmac43455-sdio.AW-CM256SM.txt -> brcmfmac43455-sdio.beagle,am5729-beagleboneai.txt
+Link: brcm/brcmfmac43455-sdio.AW-CM256SM.txt -> brcmfmac43455-sdio.pine64,pinebook-pro.txt
+Link: brcm/brcmfmac43455-sdio.AW-CM256SM.txt -> brcmfmac43455-sdio.pine64,pinephone-pro.txt
+Link: brcm/brcmfmac43455-sdio.AW-CM256SM.txt -> brcmfmac43455-sdio.pine64,quartz64-b.txt
 
 Licence: GPLv2. See GPL-2 for details.
 
-- 
2.36.1

